File: ex/prt1/Mdc.java
public class Mdc{

    int num1;
    int num2;

    public mdc(int a,int b){
        this.num1=a;
        this.num2=b;
    }

    public int calc(){
        if (this.num1==this.num2){
            return this.num1;
        }else if(this.num2>this.num1){
            this.num2-=this.num1;
            return calc();
        }else{
            this.num1-=this.num2;
            return calc();
        }
    }
}

File: ex/prt1/mdc.pl
mdc(A,A,MDC):-MDC is A.
mdc(A,B,MDC):- A>B, A1 is A-B, mdc(A1,B,MDC).
mdc(A,B,MDC):- B>A, B1=B-A, mdc(A,B1,MDC).